daemon/algod/api/server/router.go

```diff
@@ -40,6 +41,15 @@ import (
 	"github.com/algorand/go-algorand/util/tokens"
 )
 
+// APINodeInterface describes all the node methods required by common and v2 APIs, and the server/router
+type APINodeInterface interface {
+	lib.NodeInterface
+	v2.NodeInterface
+	ListeningAddress() (string, bool)
+	Start()
+	Stop()
```

Comment:
Yeah I can, but then we'll have the following 
`daemon/algod/server.go`
```go
type ServerNode interface {
	lib.NodeInterface
	v2.NodeInterface
	ListeningAddress() (string, bool)
	Start()
	Stop()
}
```
`daemon/algod/api/server/router.go`
```go
type APINodeInterface interface {
	lib.NodeInterface
	v2.NodeInterface
}
```

Since there's already a dependency from the `algod` package to the `server` package.